gangplank/spec/jobspec.go

```diff
@@ -136,8 +141,50 @@ type Job struct {
 //   GitRef: branch/ref to fetch from
 //   GitUrl: url of the repo
 type Recipe struct {
-	GitRef string `yaml:"git_ref,omitempty" json:"git_ref,omitempty"`
-	GitURL string `yaml:"git_url,omitempty" json:"git_url,omitempty"`
+	GitRef string  `yaml:"git_ref,omitempty" json:"git_ref,omitempty"`
+	GitURL string  `yaml:"git_url,omitempty" json:"git_url,omitempty"`
+	Repos  []*Repo `yaml:"repos,omitempty" json:"repos,omitempty"`
+}
+
+// Repo is a yum/dnf repositories to use as an installation source.
+type Repo struct {
+	Name string `yaml:"name,omitempty" json:"name,omitempty"`
+	URL  string `yaml:"url,omitempty" json:"url,omitempty"`
+}
+
+// Writer places the remote repo file into path. If the repo has no name,
+// then a SHA256 of the URL will be used. Returns path of the file and err.
+func (r *Repo) Writer(path string) (string, error) {
+	if r.URL == "" {
+		return "", errors.New("URL is undefined")
+	}
+	rname := r.Name
+	if rname == "" {
+		h := sha256.New()
+		if _, err := h.Write([]byte(r.URL)); err != nil {
+			return "", fmt.Errorf("failed to calculate name: %v", err)
+		}
+		rname = fmt.Sprintf("%x", h.Sum(nil))
+	}
+
+	f := filepath.Join(path, fmt.Sprintf("%s.repo", rname))
```

Comment:
Should you check to see the path exists first?

Reply:
Or rather, check before you try the `OpenFile()` below

Reply:
For the sake of being really pendantic, then yeah, I could check. However, `os.Stat` and `os.Open` both will return `os.PathError`...so I skipped it. I also figured that since this is called after `cosa init`, we'll have other problems if `src/config` doesn't exist.